File: java/src/test/java/io/binac/leetcode/MedianOfTwoSortedArraysTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MedianOfTwoSortedArraysTest {
    private static final MedianOfTwoSortedArrays.Solution1 SOLUTION1 = new MedianOfTwoSortedArrays.Solution1();

    private final int[] NUMS11 = {1, 3};
    private final int[] NUMS12 = {2};
    private final double EXPECTED1 = 2;

    private final int[] NUMS21 = {1, 2};
    private final int[] NUMS22 = {3, 4};
    private final double EXPECTED2 = 2.5;

    private final int[] NUMS31 = {0, 0};
    private final int[] NUMS32 = {0, 0};
    private final double EXPECTED3 = 0;

    private final int[] NUMS41 = {};
    private final int[] NUMS42 = {1};
    private final double EXPECTED4 = 1;

    private final int[] NUMS51 = {2};
    private final int[] NUMS52 = {};
    private final double EXPECTED5 = 2;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.findMedianSortedArrays(NUMS11, NUMS12));
        assertEquals(EXPECTED2, SOLUTION1.findMedianSortedArrays(NUMS21, NUMS22));
        assertEquals(EXPECTED3, SOLUTION1.findMedianSortedArrays(NUMS31, NUMS32));
        assertEquals(EXPECTED4, SOLUTION1.findMedianSortedArrays(NUMS41, NUMS42));
        assertEquals(EXPECTED5, SOLUTION1.findMedianSortedArrays(NUMS51, NUMS52));
    }
}

File: cpp/tests/MedianOfTwoSortedArraysTest.cpp
#include "MedianOfTwoSortedArrays.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(MedianOfTwoSortedArrays, Solution4_1) {
  auto S4_1 = Solution4_1();

  std::vector<int> Nums11{1, 3}, Nums12{2};
  const double Expected1 = 2;
  EXPECT_DOUBLE_EQ(Expected1, S4_1.findMedianSortedArrays(Nums11, Nums12));

  std::vector<int> Nums21{1, 2}, Nums22{3, 4};
  const double Expected2 = 2.5;
  EXPECT_DOUBLE_EQ(Expected2, S4_1.findMedianSortedArrays(Nums21, Nums22));

  std::vector<int> Nums31{0, 0}, Nums32{0, 0};
  const double Expected3 = 0;
  EXPECT_DOUBLE_EQ(Expected3, S4_1.findMedianSortedArrays(Nums31, Nums32));

  std::vector<int> Nums41{}, Nums42{1};
  const double Expected4 = 1;
  EXPECT_DOUBLE_EQ(Expected4, S4_1.findMedianSortedArrays(Nums41, Nums42));

  std::vector<int> Nums51{2}, Nums52{};
  const double Expected5 = 2;
  EXPECT_DOUBLE_EQ(Expected5, S4_1.findMedianSortedArrays(Nums51, Nums52));
}